### PR TITLE
[2.6] Start k3s with embedded etcd

### PR DESCRIPTION
**Pull request by annablender**
Issue [33850](https://github.com/rancher/rancher/issues/33850). Removes etcd binary and starts Rancher single node k3s cluster with embedded etcd.

#### Design plan

- Rip etcd binary and env vars out of rancher/rancher Dockerfile.dapper and package/Dockerfile
- Update the Norman API
    - Remove handler function RunETCD that starts etcd
    - Remove —datastore-endpoint
    - Start k3s server with —cluster-init (which will use embedded etcd)
    
#### What was fixed, or what changes have occurred

- Remove etcd.go etcd_linux.go and runETCD function that runs the old etcd binary
- Add --cluster-init flag to k3s server cmd
- Update no-deploy deprecated flags

#### Areas or cases that should be tested

See test plan in https://github.com/rancher/rancher/pull/35323.

### CODE REVIEW

File: pkg/kwrapper/k8s/k3s_linux.go
Comment (nickgerace) [inline pkg/kwrapper/k8s/k3s_linux.go:38]: nit: I think we can remove this comment since this PR and attached issue will indicate this as such. Perhaps, this would be better for a PR comment or a link to the k3s docs.

**Review by nickgerace — APPROVED** [review]

File: pkg/kwrapper/k8s/k3s_linux.go
Comment (brandond) [reply pkg/kwrapper/k8s/k3s_linux.go:38]: +1 on this nit. Will approve after it's resolved.

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/kwrapper/k8s/k3s_linux.go
Comment (brandond) [inline pkg/kwrapper/k8s/k3s_linux.go:44]: The no-deploy flag is deprecated, can you replace this with:
```suggestion
		"--disable=traefik,servicelb,metrics-server,local-storage",
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/kwrapper/k8s/k3s_linux.go
Comment (annablender) [reply pkg/kwrapper/k8s/k3s_linux.go:43]: Yep glad I brought that up! Will do

**Review by nickgerace — APPROVED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Comment by nickgerace** [comment]
Once approval comes from @brandond, you are good to merge @annablender.

**Review by brandond — APPROVED** [review]

**Review by kinarashah — APPROVED** [review]

**Review by genexpr — APPROVED** [review]

**Comment by nickgerace** [comment]
@kinarashah may you merge this due to permissions?

**Pull request merged**
